File: 03-Refrescos/src/main/java/es/joseluisgs/dam/model/Refresco.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package es.joseluisgs.dam.model;

import java.io.Serializable;

/**
 *
 * @author link
 */
public class Refresco implements Serializable {
    private String nombre;

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    public Refresco(){
        switch((int) (Math.random() * 7)){
            case 0: nombre = "Coca Cola";
            break;
            case 1: nombre = "Sprite";
            break;
            case 2: nombre = "Fanta";
            break;
            case 3: nombre = "Tónica";
            break;
            case 4: nombre = "Nestea";
            break;
            case 5: nombre = "Cerveza";
            break;
            case 6: nombre = "Zumo";
            break;
            default: nombre = "Agua";
        }
    }
}

File: 11-Refrescos-Kotlin/src/main/kotlin/model/Refresco.kt
package es.joseluisgs.dam.clienteservidor.model

import java.io.Serializable
import kotlin.random.Random

data class Refresco(
    var nombre: String = "Agua",
    val precio: Int = (1..3).random()
) : Serializable {
    init {
        nombre = when ((Math.random() * 7).toInt()) {
            0 -> "Coca Cola"
            1 -> "Sprite"
            2 -> "Fanta"
            3 -> "Tónica"
            4 -> "Nestea"
            5 -> "Cerveza"
            6 -> "Zumo"
            else -> "Agua"
        }
    }
}
